ls1intum/Artemis
# `Communication`: Configure messaging code of conduct

### Checklist
#### General


#### Server
#### Client

### Motivation and Context
https://github.com/ls1intum/Artemis/issues/7085: Write and edit a course's code of conduct

### Description
Add a markdown editor to the update course component and persist the code of conduct.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 course

1. Log in to Artemis
2. Navigate to course administration
3. Select a course
4. Edit the course
5. Enable messaging
6. Write a code of conduct
7. Save
8. Edit again, see that Artemis persists the code of conduct

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage
#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [course-update.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5355/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/manage/course-update.component.ts.html) | 81.01% | ✅ |
| [course.model.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5355/latest/artifact/shared/Coverage-Report-Client-Tests/app/entities/course.model.ts.html) | 100% | ✅ |
| [course-conversations.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5355/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/course-conversations/course-conversations.component.ts.html) | 100% | ✅ |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [Course.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5355/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain/Course.html) | 92% | ✅|

### Screenshots


<img width="1733" alt="Screenshot 2023-09-07 at 17 14 11" src="https://github.com/ls1intum/Artemis/assets/23535000/44096a1b-b63a-4a73-9375-05d4337ae8ff">
<img width="1733" alt="Screenshot 2023-09-07 at 17 14 23" src="https://github.com/ls1intum/Artemis/assets/23535000/3340cba2-7e2a-4590-99e2-0ded07f0670e">
